### PR TITLE
fix: runtime test for mac

### PR DESCRIPTION
**Pull request by xingyaoww**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

Fix https://github.com/OpenDevin/OpenDevin/issues/2992.

---

**Give a summary of what the PR does, explaining any non-trivial design decisions**

- Move `use_host_network` to `SandboxConfig` so it is easier to support in the new Runtime & adjust documentations accordingly
- Add support for `EventStreamRuntime` to run without `use_host_network`
- Add instruction for running `use_host_network` on Mac OS: You need to upgrade to the latest docker desktop and enable that feature manually - https://docs.docker.com/network/drivers/host/#docker-desktop
- Make `test_runtime.py` tests running without `use_host_network` to avoid issues.

---
**Other references**

### CODE REVIEW

**Review by iFurySt — APPROVED** [review]

**Comment by iFurySt** [comment]
LGTM. Thanks!👍

File: opendevin/runtime/client/runtime.py [hunk 1]
```diff
@@ -127,7 +136,7 @@ async def _init_container(
                     f'--working-dir {sandbox_workspace_dir} '
                     f'--plugins {plugin_names}'
                 ),
-                network_mode='host',
+                **network_kwargs,
```
Comment (neubig) [inline opendevin/runtime/client/runtime.py:139]: Kwargs reduce legibility a bit, so it might be better to just use arguments?

**Review by neubig — COMMENTED** [review]
Looks great! Can we undo #2993 also now that we have a real fix?

File: opendevin/runtime/client/runtime.py (hunk 1 above)
Comment (xingyaoww) [reply opendevin/runtime/client/runtime.py:139]: I guess the tricky thing is that we have two different key arguments for different scenarios, which might complicate things a bit.

Plus, we have a separate workflow dedicated for running `test_runtime` (it needs more space for docker images?), so i think we might leave #2993 as is (unless we want to run `test_runtime` together with other testcases).

Comment (neubig) [reply opendevin/runtime/client/runtime.py:139]: Ah, got it. Thanks!

Comment (neubig) [reply opendevin/runtime/client/runtime.py:139]: In terms of the different key arguments, it seems that we could pass in `None` when arguments are not used, but doing it with kwargs is not a huge deal either.

Comment (xingyaoww) [reply opendevin/runtime/client/runtime.py]: Just fixed kwargs in the latest commit!

**Comment by tobitege** [comment]
@xingyaoww could you take a peek at this action log error, please (and the ones below that line)?
https://github.com/OpenDevin/OpenDevin/actions/runs/10002545326/job/27648010988?pr=2602#step:7:2162
```
04:34:23 - opendevin:INFO: runtime_build.py:205 - Error pulling image od_runtime:ubuntu_tag_22.04, building it from scratch
04:34:23 - opendevin:INFO: runtime_build.py:206 - Non-fatal error: 404 Client Error for http+docker://localhost/v1.45/images/create?tag=ubuntu_tag_22.04&fromImage=od_runtime: Not Found ("pull access denied for od_runtime, repository does not exist or may require 'docker login': denied: requested access to the resource is denied")
```
Should we upload at least one dummy "od_runtime" image to prevent this `Client Error`, thus eliminating the constant rebuild of the image for each test in the script?
Running the actual build process in this test should only be required once, right?

**Comment by neubig** [comment]
Good idea @tobitege , I opened an issue: https://github.com/OpenDevin/OpenDevin/issues/3015

**Comment by xingyaoww** [comment]
@tobitege yep! I plan to add a CI to pre-build this image and push to ghcr - still figuring out a good way to version it!

**Comment by tobitege** [comment]
We need to find a way to reduce workflows that create images left and right for each and every one-line commit, if possible.
But I don't know if that's possible. The workflows just take too long in total, imho.

**Comment by xingyaoww** [comment]
@tobitege i'm thinking about just doing "once" per release - since this is only needed for user. For development / CI testing, we will just update the code in that container and will not rebuild it from scratch

**Comment by tobitege** [comment]
it's offtopic, I know, but mayb a 
_as long as a PR doesn't touch the "opendevin" or "tests" folder, no image creation required_
sortof rule?

**Pull request merged**